hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/FileIOEngine.java

```diff
@@ -59,6 +65,30 @@ public FileIOEngine(long capacity, String... filePaths) throws IOException {
     this.fileChannels = new FileChannel[filePaths.length];
     this.rafs = new RandomAccessFile[filePaths.length];
     this.channelLocks = new ReentrantLock[filePaths.length];
+    init();
+  }
+
+  @Override
+  public boolean verifyFileIntegrity(byte[] persistentChecksum, String algorithm)
+    throws IOException {
+    try {
+      byte[] calculateChecksum = calculateChecksum(algorithm);
+      if (!Bytes.equals(persistentChecksum, calculateChecksum)) {
+        throw new IOException("The persistent checksum is " + Bytes.toString(persistentChecksum) +
```

Comment:
The response from this method is in 2 ways for IOE or mismatch of checksum. In caller place we check for boolean return value basically.  Can we just log the issue here and return boolean in any case?